src/hotspot/cpu/ppc/interp_masm_ppc_64.cpp

```diff
@@ -681,13 +681,13 @@ void InterpreterMacroAssembler::unlock_if_synchronized_method(TosState state,
     // Check if synchronized method or unlocking prevented by
     // JavaThread::do_not_unlock_if_synchronized flag.
     lbz(Rdo_not_unlock_flag, in_bytes(JavaThread::do_not_unlock_if_synchronized_offset()), R16_thread);
-    lwz(Raccess_flags, in_bytes(Method::access_flags_offset()), R19_method);
+    lhz(Raccess_flags, in_bytes(Method::access_flags_offset()), R19_method);
     li(R0, 0);
     stb(R0, in_bytes(JavaThread::do_not_unlock_if_synchronized_offset()), R16_thread); // reset flag
 
     push(state);
 
-    // Skip if we don't have to unlock.
+    // Skip if we don't have to unlock. (???is this right???)
```

Comment:
The logic seems consistent with other platforms. Not sure what you are querying.

Reply:
It wasn't the logic.  When I went through I didn't know if this instruction needed fixing because we loaded an unsigned short instead of an int.  So I left myself a note to look at it again that you noticed and I didn't in my final walk through.  It seems right but maybe someone with ppc knowledge can answer this.

```
    rldicl_(R0, Raccess_flags, 64-JVM_ACC_SYNCHRONIZED_BIT, 63); // Extract bit and compare to 0.
```

Reply:
The instruction looks still correct. We are checking the same bit of the 64 bit register as before. (Using `testbitdi` would also work.)